src/components/StepsList/index.jsx

```diff
@@ -99,36 +90,32 @@ const StepsList = ({
     setVisible(displayToolTip());
   }, [activeStep, availableSteps, hovering]);
 
+  const handleStepClick = (index) => {
+    setActiveStep(index);
+  };
+
   return (
     <Box sx={{ color: 'primary.text' }}>
-      <Stepper activeStep={activeStep} alternativeLabel nonLinear>
-        {calculatorList.map((label, index) => (
-          <Step
-            key={label}
-            disabled={maxAvailableStep + 1 < index}
-            sx={{
-              '& .MuiSvgIcon-root': {
-                color: maxAvailableStep + 1 < index ? '' : '#4f5f30',
-                '&.Mui-active': {
-                  color: '#77b400',
-                },
+      <PSAStepper
+        steps={calculatorList}
+        strokeColor="#fffff2"
+        maxAvailableStep={maxAvailableStep}
```

Comment:
I think there are some logic problems with `maxAvailableStep`, for example, when I go to step 4 and go back, the `maxAvailableStep` should be step 4 but currently it'll become step 3